frontend/src/components/Summary/ImplementationRow.jsx

```diff
@@ -4,21 +4,40 @@ import { useState } from "react";
 import { DetailsButtonModal } from "../Modals/DetailsButtonModal";
 import { RunTimeInfoModal } from "../Modals/RunTimeInfoModal";
 import { mapLanguage } from "../../data/mapLanguage";
+import { NavLink } from "react-router-dom";
 
 const ImplementationRow = ({ cases, implementation }) => {
   const [showDetails, setShowDetails] = useState(false);
   const [showRunTimeInfo, setShowRunTimeInfo] = useState(false);
 
+  const getPath = (language, name) => {
+    const urlPath = {
+      "c++-valijson": "cpp-valijson",
```

Comment:
Cool! I'll take a closer look at the entire PR, but because I see it immediately -- I wouldn't want us hardcoding anything here (or really in the UI in general).

I believe which implementations support draft2020-12 is already available in the data outputted by the 2020 report (but if that or anything else ever wasn't available, we should make it available rather than hard coding it, we control both sides here!)

Reply:
Yes, you are absolutely correct. The data is, indeed, available within the report. However, to grab the data related to a specific implementation, we need to verify whether the **name** and **language** of that implementation are present in any of the available image URLs.

In most cases, the Image URL follows the pattern `/<language>-<name>` at the end, for example `go-jsonschema`. However, there are cases where obtaining data for certain implementations becomes challenging due to their names. For example, matching `JsonSchema.Net (dotnet)` with `/dotnet-jsonschema-net` is difficult. Due to this reason, I've hard-coded these specific values.

If we want to make the entire path dynamic, we would need to make slight adjustments to the implementation names. For example, in the metadata, we would have to change the language name from `c++` to `cpp`, implementation name form `JsonSchema.Net` to `jsonschema-net` and so on.

Reply:
Yes, you are absolutely correct. The data is, indeed, available within the report. However, to grab the data related to a specific implementation, we need to verify whether the **name** and **language** of that implementation are present in any of the available image URLs.

In most cases, the Image URL follows the pattern `/<language>-<name>` at the end, for example `go-jsonschema`. However, there are cases where obtaining data for certain implementations becomes challenging due to their names. For example, matching `JsonSchema.Net (dotnet)` with `/dotnet-jsonschema-net` `(ghcr.io/bowtie-json-schema/dotnet-jsonschema-net)` is difficult. Due to this reason, I've hard-coded these specific values.

If we want to make the entire path dynamic, we would need to make slight adjustments to the implementation names. For example, in the metadata, we would have to change the language name from `c++` to `cpp`, implementation name form `JsonSchema.Net` to `jsonschema-net` and so on.

Reply:
Looking at the report, from what I see, this is all already there. This is the first line for a run I just did with 1 implementation:

```json
{"started": "2023-08-14T17:54:15.968585+00:00", "bowtie_version": "2023.8.2.dev53+g5dc9de3.d20230804", "dialect": "https://json-schema.org/draft/2020-12/schema", "implementations": {"ghcr.io/bowtie-json-schema/python-jsonschema": {"language": "python", "name": "jsonschema", "version": "4.18.4", "homepage": "https://python-jsonschema.readthedocs.io/", "issues": "https://github.com/python-jsonschema/jsonschema/issues", "dialects": ["https://json-schema.org/draft/2020-12/schema", "https://json-schema.org/draft/2019-09/schema", "http://json-schema.org/draft-07/schema#", "http://json-schema.org/draft-06/schema#", "http://json-schema.org/draft-04/schema#", "http://json-schema.org/draft-03/schema#"], "os": "Linux", "os_version": "6.4.7-200.fc38.aarch64", "language_version": "3.11.4", "image": "ghcr.io/bowtie-json-schema/python-jsonschema"}}, "metadata": {"Commit": {"text": "5cc9214e82f1e0a5e9644960b6fe0166afb7b283", "href": "https://github.com/json-schema-org/JSON-Schema-Test-Suite/commit/5cc9214e82f1e0a5e9644960b6fe0166afb7b283"}}}
```

which already contains the mapping between ID (image name) and language and name.

Have a look and point out if you see something specific missing.

Reply:
(`http get https://bowtie.report/draft2020-12.json | head -n1 | jq` is the "production" file obviously, and seems to already contain all this data for all implementations we run -- so you want to parse that and create your routes probably)

Reply:
To visit an implementation page, one has to click the link on the homepage. Now according to the current code, the link will take them to `base/implementations/${language}-${name}`. 
For example, when someone clicks 
![image](https://github.com/bowtie-json-schema/bowtie/assets/113545196/b6b804a9-5c01-42c5-a313-d56ed5a869ad)
then they will be redirected to `base/implementations/go-jsonschema` and when someone clicks 
![image](https://github.com/bowtie-json-schema/bowtie/assets/113545196/9fc94574-053a-49dc-9054-8020b84dacd0)
they will be redirected to `base/implementations/dotnet-JsonSchema.Net`.

In the case of `base/implementations/go-jsonschema`, the substring 'go-jsonschema' is present in one of our unique IDs (i.e. - `ghcr.io/bowtie-json-schema/go-jsonschema`), so we can use the value mapped to this image URL and display it on the following page `base/implementations/go-jsonschema`. 

But in the case of `base/implementations/dotnet-JsonSchema.Net`, it is difficult to get the value mapped to `ghcr.io/bowtie-json-schema/dotnet-jsonschema-net` ID because the image URL does not include the substring "dotnet-JsonSchema.Net" (it includes "dotnet-jsonschema-net").

However, if we could somehow visit `base/implementations/dotnet-jsonschema-net` instead of  `base/implementations/dotnet-JsonSchema.Net`, then we could use the **_dotnet-jsonschema-net_** substring to grab this `ghcr.io/bowtie-json-schema/dotnet-jsonschema-net` ID and use the associated value to display on the dotnet specific page.

Only a few of the implementations have the above-mentioned issue while the rest of them are working totally fine with `base/implementations/${language}-${name}`

Do you have any other ideas for achieving the above functionality?

Reply:
Hmm, I overlooked that the data is already available in `ImplementationRow.jsx` with a unique ID.
Fixed!